### PR TITLE
Create Partner view for Bank page

### PR DESCRIPTION
**Pull request by exu3**
Bank partners (like Givebutter for example) don't have the same features as regular Bank users and the signup flow is slightly different. This PR creates a filtered view for users coming from the Givebutter help center. https://hackclub.com/bank?ref=gb_help_desk

### CODE REVIEW

**Comment by lachlanjc** [comment]
#### 📝 Changed routes:


| [**/bank**](https://v3-38kjqveeh.hackclub.dev/bank) |
|:-:|
|<a href="https://v3-38kjqveeh.hackclub.dev/bank">Screenshot of /bank</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/uVTuHqHNSjxOEfYgw_00Bg-Reobv_Cds0fGaprUWtcsrcaoVFJ7lLFiFrDKsZsFSSZZZwI9nfV97LF0fyv74_A.png">(view full size)</a> |







Commit eb563e9a421866966848116b74d1e33c062648c7 (https://v3-38kjqveeh.hackclub.dev).

**Comment by sampoder** [comment]
@eilla1 I've made a couple changes, mainly removing the separate partner components as I want to keep things together to make the site easier to maintain and the code more compact. Hope that's alright with you.

File: pages/bank.js [hunk 1]
```diff
@@ -35,17 +37,23 @@ export default function Bank() {
...
+          )}
+          {!isGivebutter && <Start />}
         </Box>
       </Box>
       <Footer dark key="footer" />
     </>
   )
 }
+
+export default Bank
```
Comment (sampoder) [inline pages/bank.js]: Any reason for this change? It's cool with me but just curious

**Review by sampoder — APPROVED** [review]

**Comment by cjdenio** [comment]
If you'd like to, you could server-side render this page with
```diff
-function Bank() {
-  const { ref } = useRouter().query
-  const isGivebutter = ref === 'gb_help_desk'
+function Bank({ isGivebutter }) {
   return (
     <>
       <Box as="main" key="main">
```
```diff
export default Bank

+export function getServerSideProps(context) {
+  return {
+    props: {
+      isGivebutter: context.query.ref === 'gb_help_desk'
+    }
+  }
+}
```

(might be better for performance)

File: pages/bank.js (hunk 1 above)
Comment (exu3) [reply pages/bank.js]: No reason, I was just experimenting but forgot to change it back.

**Comment by exu3** [comment]
> @eilla1 I've made a couple changes, mainly removing the separate partner components as I want to keep things together to make the site easier to maintain and the code more compact. Hope that's alright with you.

Thanks Sam! I think that makes it cleaner.

**Comment by exu3** [comment]
> If you'd like to, you could server-side render this page with

Wahoo I like this idea. I think Sam wanted to keep it static though https://github.com/hackclub/site/pull/310#commitcomment-64297681. Thoughts on making this page server-side rendered?

I'm going to go ahead and merge as is in the interest of getting it live.

**Comment by cjdenio** [comment]
It's actually not static currently (you can't statically optimize pages based on query parameters).

Because the query parameter is being checked on the client-side, you get this brief flash of an "Apply now" button:

https://user-images.githubusercontent.com/34525547/150583514-d6eb37c0-1c4a-4655-acdc-21833850eab1.mov

**Pull request merged**